### DISCUSSION TITLE
No Signal K server found in network when using mDNS service!

### DISCUSSION THREAD

**Discussion by zschramm** [SignalK/SensESP#686] [General]
HI,
Getting this error when trying to get connected to Signal K.  I tried both hardcoding the server and letting it search for server, and also with server specified for both port 3000 and 3443.

I got this working at my office without specifying the server, so I do know it works.  On the boat I have a CerboGX hosting signal K, but HTTP access is on port 3000 and I connect over SSL generally on 3443 (even though I don't have real cert).

'
(load_configuration)(C1) Could not find configuration for path /System/WiFi Settings
(load_configuration)(C1) Loading configuration for path '/System/Signal K Settings' from '/7NAVR03_FL3ag9_pt1mBQlpDmeM='
(load_configuration)(C1) Not loading configuration: no config_path specified: 
(load_configuration)(C1) Not loading configuration: no config_path specified:
(SKOutputNumeric)(C1) WARNING - No metadata for propulsion.0.exhaustTemperature. Numeric values should specify units
(load_configuration)(C1) Not loading configuration: no config_path specified:
(SKOutputNumeric)(C1) WARNING - No metadata for propulsion.1.exhaustTemperature. Numeric values should specify units
(start)(C1) Starting subsystems
(start_all)(C1) Starting all required sensors and transforms
(start)(C1) Enabling Networking object
(start)(C1) Using SSID Off Hand 2.4G
(operator())(C1) Connecting to wifi SSID Off Hand 2.4G.
(start)(C1) mDNS responder started for hostname 'egt-temp'
(connect)(C0) WiFi is disconnected. SignalK client connection will connect when WiFi is connected.
(start)(C1) HTTP server not started, WiFi not connected
(start)(C1) Button handler started
(start)(C1) All sensors and transforms started
(wifi_station_connected)(C1) Connected to wifi, SSID: Off Hand 2.4G (signal: -48)
(wifi_station_connected)(C1) IP address of Device: 192.168.8.198
(wifi_station_connected)(C1) Default route: 192.168.8.1
(wifi_station_connected)(C1) DNS server: 192.168.8.1
(operator())(C1) HTTP server started
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
'

Mostly get the same message, but I did get one time the 104 error whatever that is.
`(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(connect)(C0) No Signal K server found in network when using mDNS service!
(connect)(C0) Initiating websocket connection with server...
(operator())(C1) Got system status: 104
(connect)(C0) No Signal K server found in network when using mDNS service!
'

To me it doesn't seem like it tries to connect with the hardcoded server address based on what I have found from others.

SignalK has mDNS enabled.  I don't know what else might be impacting the connection.  I do have AdGuard home running but didn't find any mDNS settings on there.

Thank you all.
Zac

**Comment by zschramm** [comment]
I did some more reading on this and I think the main issue is this is running on the same host that already has a HTTP (and probably mdns) running.  The victron cerbo gx has it's own interface and network services running already and also nodered and signalk.   

When looking at the discovery process I see it looks for signalk-http on port 80 of which cerbogx is already hosting on port 80.

What I don't understand is I hardcode the signalk host and port and it doesn't seem to try to connect to it, goes straight to mdns.

**Comment by mairas** [comment]
Have you checked what services are actually being broadcast on the network? E.g. `avahi-browse --all --resolve` on Linux or the Discovery app on Mac? `_signalk-ws._tcp` is the service SensESP is looking for.

There might be some unobvious behavior depending on the config saved on the device flash. Try nuking the device by pressing the BOOT button over 5 seconds.

**Comment by zschramm** [comment]
Thank you Matti.  This opens up an area I don't know much about.

I did this on the home network in advance to test this command and I don't understand why it says failed to resolve.  Ran this command on an ubuntu virtual machine once I installed avahi-utils.

```
Failed to resolve service '_signalk-ws' of type '_tcp' in domain 'local': Invalid service type
+   eth0 IPv4 _signalk-ws                                   _tcp                 local
+   eth0 IPv4 signalk                                       _signalk-ws._tcp     local
=   eth0 IPv4 signalk                                       _signalk-ws._tcp     local
   hostname = [signalk.local]
   address = [192.168.2.164]
   port = [3000]
   txt = ["vmmsi=2222" "vname=Test" "self=urn:mrn:imo:mmsi:2222" "roles=master, main" "swvers=2.6.2" "swname=signalk-server" "txtvers=1"]
Failed to resolve service '_signalk-http' of type '_tcp' in domain 'local': Invalid service type
+   eth0 IPv4 _signalk-http                                 _tcp                 local
+   eth0 IPv4 signalk                                       _signalk-http._tcp   local
=   eth0 IPv4 signalk                                       _signalk-http._tcp   local
   hostname = [signalk.local]
   address = [192.168.2.164]
   port = [3000]
   txt = ["vmmsi=2222" "vname=Test" "self=urn:mrn:imo:mmsi:2222" "roles=master, main" "swvers=2.6.2" "swname=signalk-server" "txtvers=1"]
+   eth0 IPv4 signalk                                       Web Site             local
=   eth0 IPv4 signalk                                       Web Site             local
   hostname = [signalk.local]
   address = [192.168.2.164]
   port = [3000]
   txt = ["vmmsi=2222" "vname=Test" "self=urn:mrn:imo:mmsi:2222" "roles=master, main" "swvers=2.6.2" "swname=signalk-server" "txtvers=1"]
Failed to resolve service '_signalk-tcp' of type '_tcp' in domain 'local': Invalid service type
+   eth0 IPv4 _signalk-tcp                                  _tcp                 local
+   eth0 IPv4 signalk                                       _signalk-tcp._tcp    local
=   eth0 IPv4 signalk                                       _signalk-tcp._tcp    local
   hostname = [signalk.local]
   address = [192.168.2.164]
   port = [8375]
   txt = ["vmmsi=2222" "vname=Test" "self=urn:mrn:imo:mmsi:2222" "roles=master, main" "swvers=2.6.2" "swname=signalk-server" "txtvers=1"]
Failed to resolve service '_nmea-0183' of type '_tcp' in domain 'local': Invalid service type
+   eth0 IPv4 _nmea-0183                                    _tcp                 local
+   eth0 IPv4 signalk                                       _nmea-0183._tcp      local
=   eth0 IPv4 signalk                                       _nmea-0183._tcp      local
   hostname = [signalk.local]
   address = [192.168.2.164]
   port = [10110]
   txt = ["vmmsi=2222" "vname=Test" "self=urn:mrn:imo:mmsi:2222" "roles=master, main" "swvers=2.6.2" "swname=signalk-server" "txtvers=1"]
```


Can you clarify what you mean by the "BOOT" button?  Are you referring to the victron cerbogx?

The other question is why can't I just bypass mdns and just have it connect to the IP and port of the signalk server since I know what it is and it's set to static.

```
  sensesp_app = (&builder)
                    // Set a custom hostname for the app.
                    ->set_hostname("egt-temp")
                    // Optionally, hard-code the WiFi and Signal K server
                    // settings. This is normally not needed.
                    ->set_wifi("---------", "----------")
                    ->set_sk_server("192.168.8.10", 3000)
                    ->get_app();
```

**Comment by zschramm** [comment]
Got this working finally.
I am not sure what did it in the end or if a combinination of the the following.
Disabled AdGuard DNS on the Victron Cerbo GX and that didn't change anything.
Also mDNS was disabled in Signalk but enabling that didn't work.
Finally I disabled SSL in Signalk and then I received the access request.  I thought both would work but since on a local boat network I don't need SSL.

**Reply by mairas** [reply]
Oh yeah, SSL not working is a long-standing issue that really should get fixed. Should be easier with SensESP 3.0.
